components/cloud/aws/src/util.rs

```diff
@@ -92,21 +95,29 @@ impl Default for DefaultCredentialsProvider {
 #[async_trait]
 impl ProvideAwsCredentials for DefaultCredentialsProvider {
     async fn credentials(&self) -> Result<AwsCredentials, CredentialsError> {
-        // Prefer the web identity provider first for the kubernetes environment.
-        // Search for both in parallel.
-        let web_creds = self.web_identity_provider.credentials();
-        let def_creds = self.default_provider.credentials();
-        let k8s_error = match web_creds.await {
-            res @ Ok(_) => return res,
-            Err(e) => e,
-        };
-        let def_error = match def_creds.await {
-            res @ Ok(_) => return res,
-            Err(e) => e,
+        let mut retry_cnt = 0;
+        let res = loop {
+            // use web identity provider first for the kubernetes environment.
+            let creds = if std::env::var("AWS_WEB_IDENTITY_TOKEN_FILE").is_ok() {
+                self.web_identity_provider.credentials()
+            } else {
+                self.default_provider.credentials()
+            };
+            match creds.await {
+                res @ Ok(_) => break res,
+                Err(e) => {
+                    if retry_cnt >= 4 {
+                        break Err(CredentialsError::new(format_args!(
+                            "Couldn't find AWS credentials in sources ({}).",
+                            e.message
+                        )));
+                    }
+                }
+            };
+            retry_cnt += 1;
+            // this could happen in web identity provider, we need retry it.
+            thread::sleep(time::Duration::from_secs(retry_cnt));
```

Comment:
Please use `tokio::time::sleep`, or we may block the runtime.
```suggestion
            tokio::time::sleep(time::Duration::from_secs(retry_cnt)).await;
```